src/Discord/Parts/Embed/Embed.php

```diff
@@ -38,7 +38,104 @@ class Embed extends Part
      * {@inheritdoc}
      */
     protected $fillable = ['title', 'type', 'description', 'url', 'timestamp', 'color', 'footer', 'image', 'thumbnail', 'video', 'provider', 'author', 'fields'];
-
+    
+    /**
+     * Constructs a new instance.
+     * @param array  $embed
+     * @throws \Throwable
+     */
+    function __construct(array $embed = array()) {
+        if (!empty($embed)) {
+            $this->type = $embed['type'] ?? 'rich';
+            
+            if (!empty($embed['title'])) {
+                $this->setTitle($embed['title']);
+            }
+            
+            if (!empty($embed['author'])) {
+                $this->setAuthor(
+                    ((string) ($embed['author']['name'] ?? '')),
+                    ((string) ($embed['author']['icon_url'] ?? '')),
+                    ((string) ($embed['author']['url'] ?? ''))
+                );
+            }
+            
+            if (!empty($embed['description'])) {
+                $this->setDescription($embed['description']);
+            }
+            
+            $this->url = $embed['url'] ?? null;
+            $this->timestamp = (!empty($embed['timestamp']) ? (new \DateTime($embed['timestamp']))->getTimestamp() : null);
+            $this->color = $embed['color'] ?? null;
+            
+            if (!empty($embed['footer'])) {
+                $this->setFooter(
+                    ((string) ($embed['footer']['text'] ?? '')),
+                    ((string) ($embed['footer']['icon_url'] ?? ''))
+                );
+            }
+            
+            if (!empty($embed['image'])) {
+                $this->image = array(
```

Comment:
could you use also the short array syntax in that file?
`$this->image = [...]`

Reply:
`php-cs-fixer` will change that when it's run, so it's fine for now.